src/components/ConversationSettings/LinkShareSettings.vue

```diff
@@ -21,36 +21,49 @@
 				{{ t('spreed', 'Allow guests to join this conversation via link') }}
 			</NcCheckboxRadioSwitch>
 
-			<NcCheckboxRadioSwitch v-show="isSharedPublicly"
-				:checked="isPasswordProtectionChecked"
-				:disabled="isSaving"
-				type="switch"
-				aria-describedby="link_share_settings_password_hint"
-				@update:checked="togglePassword">
-				{{ t('spreed', 'Password protection') }}
-			</NcCheckboxRadioSwitch>
-
-			<form v-if="showPasswordField" class="password-form" @submit.prevent="handleSetNewPassword">
-				<NcPasswordField ref="passwordField"
-					:value.sync="password"
-					autocomplete="new-password"
-					check-password-strength
+			<template v-if="isSharedPublicly">
+				<NcCheckboxRadioSwitch v-if="!forcePasswordProtection"
+					:checked="isPasswordProtectionChecked"
 					:disabled="isSaving"
-					class="password-form__input-field"
-					label-visible
-					:label="t('spreed', 'Enter new password')"
-					@valid="isValid = true"
-					@invalid="isValid = false" />
-				<NcButton :disabled="isSaving || !isValid"
-					type="primary"
-					native-type="submit"
-					class="password-form__button">
-					<template #icon>
-						<ArrowRight />
-					</template>
-					{{ t('spreed', 'Save password') }}
-				</NcButton>
-			</form>
+					type="switch"
+					aria-describedby="link_share_settings_password_hint"
+					@update:checked="togglePassword">
+					{{ t('spreed', 'Password protection') }}
+				</NcCheckboxRadioSwitch>
+				<p v-if="forcePasswordProtection && isPasswordProtectionChecked" class="app-settings-section__hint">
+					{{ t('spreed', 'This conversation is password-protected. Guests need password to join') }}
+				</p>
```

Comment:
For previous conversations without password, it won't show anything. Should we show a warning?
Toggle password is also hidden

Reply:
Let's show toggle all the time, but disabled if `force_passwords` is enabled?

prop `showPasswordField` might be tweaked to true with a watcher maybe to show form, when:
- passwords are forced, but not set (together with ⚠️ warning)
- password is set (in case moderator wants to see it or reset, API should provide conversation password now)

Reply:
> Let's show toggle all the time, but disabled if force_passwords is enabled?

It was like that initially but it felt like irrelevant to show a disabled switch button.

Password are stored hashed, they cannot be restored. Currently, user needs to:
- Disable/enable password IF it is not forced
- Disable/enable guest access IF it is forced

> passwords are forced, but not set (together with ⚠️ warning)

I thought about showing a warning with "set password" button, it will set to the normal state (setting guest access when password is enforced, and the warning will be removed)